internal/controller/dnsrecord_controller.go

```diff
@@ -156,15 +156,25 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 }
 
 func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, current *v1alpha1.DNSRecord, requeueAfter time.Duration) (reconcile.Result, error) {
-	current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	if requeueAfter != noRequeueDuration {
+		current.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(requeueAfter))
+	}
 
 	if !equality.Semantic.DeepEqual(previous.Status, current.Status) {
-		updateError := r.Status().Update(ctx, current)
-		if apierrors.IsConflict(updateError) {
-			return ctrl.Result{Requeue: true}, nil
+		if updateError := r.Status().Update(ctx, current); updateError != nil {
+			if apierrors.IsConflict(updateError) {
+				return ctrl.Result{Requeue: true}, nil
+			}
+			return ctrl.Result{}, updateError
```

Comment:
in here we want to requeue if we have a conflict with no error returned. And if there is another error we want a default backoff with errors in the logs. If we make the last line to be `return ctrl.Result{RequeueAfter: requeueAfter}, updateError` it should do the same but we will have less of `if` statements around.